overlord/ifacestate/handlers.go

```diff
@@ -96,9 +96,15 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, _ *tomb.Tomb) error
 		return err
 	}
 	if len(affectedSnaps) == 0 {
-		affectedSnaps = append(affectedSnaps, snapInfo)
+		affectedSnaps = append(affectedSnaps, snapName)
 	}
-	for _, snapInfo := range affectedSnaps {
+	for _, snapName := range affectedSnaps {
+
+		snapInfo, err := snapstate.Current(task.State(), snapName)
+		if err != nil {
+			return err
+		}
+		snap.AddImplicitSlots(snapInfo)
```

Comment:
This is still a temporary hack as we should not need this (we should publish this data in the OS snap normally). The helper itself would be weird as AddImplicitSlots is the helper, we access the info objects in various ways each time.

I'd much rather remove this function entirely with the next wave of OS snap updates